containers/hydra/postbuild.py

```diff
@@ -171,26 +173,17 @@ def main():
         print(f'POSTBUILD_INFO="{nixbuildinfo}"')
         print(f'POSTBUILD_INFO_SIGNATURE="{create_signature(nixbuildinfo)}"')
 
-        # generate provenance
-        result = subprocess.run(
-            [
-                "/setup/provenance.sh",
-                niximglink,  # image path
-                nixbuildinfo,  # buildinfo path
-                tmpdir,  # working dir for provenance and sbom
-                provenancename,  # file to save the provenance to
-            ],
-            stdout=subprocess.PIPE,
-            check=False
-        )
+    # Run provenance generation if script is given
+    if PROVENANCE_SCRIPT is not None:
+        result = subprocess.run(PROVENANCE_SCRIPT, stdout=subprocess.PIPE, check=False)
```

Comment:
Here you could just use os.system -command to run the script. (see how messaging script is run)
With this particular subprocess command stderr is not properly captured and if anything goes wrong there will be python exception, without information what actually went wrong with the provenance script. 
I admit, that this is a bug that I have introduced originally in running nix-store command, and you copied it. My upcoming changes will fix this on nix-store part.
Anyway with os.system -command stdout goes to stdout and stderr goes to stderr and you don't have worry about explicitly printing the provenance script outputs.